modules/framework/galasa-parent/dev.galasa.framework/src/main/java/dev/galasa/framework/internal/streams/StreamsServiceImpl.java

```diff
@@ -120,4 +120,23 @@ private List<IStream> handleStreamProperties(String testStreamPrefix) throws Con
 
         return streamsList;
     }
+
+    @Override
+    public void deleteStream(String streamName) throws StreamsException {
+        
+        String testStreamPrefix = TEST_STREAM_PREFIX + streamName + ".";
+        try {
+            // Get all the properties with the streamName
+            Map<String, String> testStreamProperties = cpsService.getPrefixedProperties(testStreamPrefix);
+
+            for (Map.Entry<String,String> entry : testStreamProperties.entrySet()) {
+                // Delete all the properties
+                cpsService.deleteProperty(entry.getKey());
+            }
+
+        } catch (ConfigurationPropertyStoreException e) {
+            e.printStackTrace();
```

Comment:
Throw an exception instead of printing the stack trace here?

What happens if one of the test stream properties fails to be deleted, so we're left with a partially-deleted stream? Can we still delete the remaining properties? I think deleting should be an all-or-nothing operation